source/slang/slang-options.cpp

```diff
@@ -1689,21 +1943,50 @@ struct OptionsParser
                     artifact->addRepresentation(fileRep);
 
                     SLANG_RETURN_ON_FAIL(_addLibraryReference(requestImpl, artifact));
+                    break;
                 }
-                else if (argValue == "-v" || argValue == "-version")
+                case OptionKind::Version:
                 {
                     sink->diagnoseRaw(Severity::Note, session->getBuildTagString());
+                    break;
                 }
-                else if (argValue == "-h" || argValue == "-help" || argValue == "--help")
+                case OptionKind::Help:
```

Comment:
We should handle `--help` separately, i.e. if any option is `--help` then print the help text and exit(0). This enables one to do `slangc --invalid option --help` and still get something useful. 

See also: https://github.com/shader-slang/slang/issues/2829

Reply:
Will look to fix in later PR.